### PR TITLE
Fix TextField losing focus when help text is added or removed and no label is provided

### LINKED ISSUE #3473: A TextField/ComboBox without label and description loses focus when the errorMessage is shown or hidden.
# 🐛 Bug Report 

A TextField component that doesn't have label and description loses focus when the errorMessage is shown or hidden.

## 🤔 Expected Behavior

The component should not lose focus.

## 😯 Current Behavior

Focus is lost when the error message is shown or hidden.

## 💁 Possible Solution

If you don't plan to fix this issue very quickly, maybe you can add a warning in documentation or in the console log. 

## 🔦 Context

It is very annoying to lose focus of the input while you are typing . I lost 2 days debugging my application until I found the problem.

This error was first reported 9 months ago, but no bug was logged at that time. I think other people will be affected by this and my suggestion would be to raise the priority of this bug.

## 💻 Code Sample

https://codesandbox.io/s/react-spectrum-input-error-problem-6ftlye

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.21.1
| Browser          | Chrome
| Operating System | macOS Monterey

## 🧢 Your Company/Team

Adobe RTCDP Connections

### PR DESCRIPTION
**Pull request by reidbarber**
Closes #3473

Previously, when there was no visible label and help text was added/removed as the user typed, the text input would lose focus.

This is because there were two separate render trees. The one for the case of no help text and no label did not include a wrapper div. This PR changes updates this to always include the wrapper div, so focus won't move when help text is added/removed.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Test new story at HelpText -> Aria Label with Dynamic Help Text.

Input should remain focused as the user types and help text gets added or removed.

### CODE REVIEW

File: packages/@react-spectrum/label/src/Field.tsx [hunk 1]
```diff
@@ -50,125 +50,108 @@ function Field(props: SpectrumFieldProps, ref: RefObject<HTMLElement>) {
...
-        {...wrapperProps}
-        ref={ref as RefObject<HTMLDivElement>}
-        className={labelWrapperClass}>
-        {labelAndContextualHelp}
-        {renderChildren()}
       </div>
     );
   }
 
-  return React.cloneElement(children, mergeProps(children.props, {
```
Comment (reidbarber) [inline packages/@react-spectrum/label/src/Field.tsx:160]: Diff is not very clean, but it's basically just removing this case.

**Review by snowystinger — COMMENTED** [review]
I ran chromatic, lets talk about this during Monday's testing

**Review by snowystinger — APPROVED** [review]
New story works as advertised, changes look good
We do have technically one more place where focus could be lost, if the labelPosition changes from side <-> top, but I don't think that's a common use case while focus is inside the textfield
nor do I think it'd be common to restore focus to a field that went switched label position, but something to keep in mind.

**Review by ktabors — DISMISSED** [review]
Changes work and they look good.

**Review by snowystinger — DISMISSED** [review]

**Comment by snowystinger** [comment]
The input is quite jumpy in the new story, is that something we should worry about in a real life application?

**Comment by reidbarber** [comment]
> The input is quite jumpy in the new story, is that something we should worry about in a real life application?
> 
> If a width is placed on it or it's flexed?

The height change is expected with how we `align-items: center` everything in storybook. The width jumping was actually a bug in our CSF3 story. We were providing argTypes `width` with a defaultValue of 'top', which isn't an option, so it was automatically using a width. I don't think we want any deafultValue's for our controls. Just updated that story.

**Review by snowystinger — APPROVED** [review]

**Review by ktabors — APPROVED** [review]

**Pull request merged**